pwa/app/components/tba/auth/signInButton.tsx

```diff
@@ -0,0 +1,41 @@
+import { AuthProvider, signInWithPopup } from 'firebase/auth';
+
+import { auth } from '~/firebase/firebaseConfig';
+
+interface SignInButtonProps {
+  provider: AuthProvider;
+  logo: string;
+  buttonText: string;
+  buttonStyles: string;
+}
+
+const SignInButton = ({
+  provider,
+  logo,
+  buttonText,
+  buttonStyles,
+}: SignInButtonProps) => {
+  const handleSignIn = () => {
+    signInWithPopup(auth, provider)
+      .then((result) => {
+        console.log('User signed in:', result.user);
+      })
+      .catch((error: unknown) => {
+        console.error('Error during sign-in:', error);
+      });
+  };
+
+  return (
+    <button
+      className={`w-full flex items-center justify-center h-10 text-base rounded ${buttonStyles} cursor-pointer`}
```

Comment:
Tailwind doesn't allow embedding styles via string interpolation. Try this:

`className={cn('w-full flex items-center ...', buttonStyles)}`

`cn` is a helper function that calls `clsx` which merges things in a tailwind compliant manner. I don't know enough about how it works to explain it